### PR TITLE
[MHLO] DisposableElementsAttr to mhlo,  ConstProp for Equal and ConstantOfShape

### PR DESCRIPTION
**Pull request by chenchongsong**
- Converts ONNX Dialect with DisposableElementsAttr to Mhlo
- Supports ConstProp for Equal and ConstantOfShape Op

@AlexandreEichenberger @tungld @yaochengji Please kindly have a look :)

### CODE REVIEW

File: src/Transform/ONNX/ConstProp.cpp
Comment (sorenlassen) [inline src/Transform/ONNX/ConstProp.cpp:864]: please add a comment here or in ConstProp.td explaining that we rely on value always being set, namely by the (questionable) "canonicalization" in ONNXConstantOfShapeOp::inferShapes()

(the casual reader my wonder about this given that the onnx spec says that the value attribute is optional)

File: src/Transform/ONNX/ConstProp.cpp
Comment (sorenlassen) [inline src/Transform/ONNX/ConstProp.cpp:862]: I think you can do this simpler:
```
llvm::SmallVector<int64_t, 4> shapeVector(shapeAttr.getValues<int64_t>());
```

File: src/Transform/ONNX/ConstProp.cpp
Comment (sorenlassen) [inline src/Transform/ONNX/ConstProp.cpp:867]: if shapeVector is empty (the result is a scalar) then this stretches the functionality of expand because we "contract" constElements's 1D shape to 0D

looking at the implementation of expand() it's possible that it could work but can you please add a lit test for this? and maybe it's best to special case this with something like:
```
if (shapeVector.empty()) expandedElements = elementsBuilder.reshape(constElements, shapeVector);
```

**Comment by tungld** [comment]
@chenchongsong thanks for adding ConstProp for Equal and ConstantOfShape! Please make changes according to the comments from @sorenlassen.

**Comment by chenchongsong** [comment]
Thanks for the review from @sorenlassen ! These comments are very helpful. Changes have been made accordingly. Could you have a look again? @sorenlassen @tungld

**Comment by tungld** [comment]
@jenkins-droid test this please

File: src/Conversion/ONNXToMhlo/Tensor/Constant.cpp
Comment (tungld) [inline src/Conversion/ONNXToMhlo/Tensor/Constant.cpp:16]: Do we still need to include this header file?

**Review by tungld — APPROVED** [review]
LGTM. Thanks for the patch!

Please wait for @sorenlassen's comments before merging this.

**Review by sorenlassen — APPROVED** [review]
LGTM, thanks for making the changes

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
Will merge this once all required Jenkins CIs passed. Thanks!

**Pull request merged**
